bats/tests/k8s/spinkube-npm.bats

```diff
@@ -0,0 +1,87 @@
+load '../helpers/load'
+
+local_setup_file() {
+    echo "$RANDOM" >"${BATS_FILE_TMPDIR}/random"
+}
+
+local_setup() {
+    if using_docker; then
+        skip "this test only works on containerd right now"
+    fi
+    if ! command -v "npm${EXE}" >/dev/null; then
+        skip "this test requires npm${EXE} to be installed and on the PATH"
+    fi
+    needs_port 80
+
+    MY_APP=my-app
+    MY_APP_NAME="${MY_APP}-$(cat "${BATS_FILE_TMPDIR}/random")"
+    MY_APP_IMAGE="ttl.sh/${MY_APP_NAME}:15m"
```

Comment:
Not really happy about using a random service for a temporary registry; can we get away with having it pre-loaded into containerd? (That is, skip the `spin registry push`)

Reply:
The idea behind the test is to go through the whole app dev/deployment lifecycle, so I don't want to skip the step.

We could deploy our own registry (like `test/registry/creds.bats`), if we want to, but I think `ttl.sh` is fine for now.  It is hosted by Cloudflare, and seems to be used by most of the Wasm blog posts and tutorials. It also keeps the test small/simple.[^1]

[^1]: We could of course pull out all the registry handling code from `creds.bats` into a `helper/registry.bash` library.

So I would leave it for now, and if/when we include a bundled registry that we can enable with a setting, then we may switch to that in the future.